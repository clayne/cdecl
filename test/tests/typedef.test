EXPECTED_EXIT=0
cdecl -O <<END
typedef int I, *PI, A2[2], F_I(double), *F_PI(), (*PF_I)()
typedef F_I *PF_I_2
typedef void Void
show as english

declare x as I
explain I x

typedef int int_least32_t
END
