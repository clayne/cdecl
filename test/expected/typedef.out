cdecl> typedef int I, *PI, A2[2], F_I(double), *F_PI(), (*PF_I)()
cdecl> typedef F_I *PF_I_2
cdecl> typedef void Void
cdecl> show as english
define A2 as array 2 of integer
define F_I as function (double precision) returning integer
define F_PI as function returning pointer to integer
define I as integer
define PF_I as pointer to function returning integer
define PF_I_2 as pointer to F_I
define PI as pointer to integer
define Void as void
cdecl> declare x as I
I x;
cdecl> explain I x
declare x as I
cdecl> typedef int int_least32_t
